extensions/charts/payloads.go

```diff
@@ -80,14 +80,14 @@ func newChartInstall(name, version, clusterID, clusterName, url, repoName, proje
 }
 
 // newChartUpgradeAction is a private constructor that creates a chart upgrade with given chart values that can be used for chart upgrade action.
-func newChartUpgrade(name, version, clusterID, clusterName, url, defaultRegistry string, chartValues map[string]interface{}) *types.ChartUpgrade {
+func newChartUpgrade(chartName, releaseName, version, clusterID, clusterName, url, defaultRegistry string, chartValues map[string]interface{}) *types.ChartUpgrade {
 	chartUpgrade := types.ChartUpgrade{
 		Annotations: map[string]string{
 			"catalog.cattle.io/ui-source-repo":      "rancher-charts",
 			"catalog.cattle.io/ui-source-repo-type": "cluster",
 		},
-		ChartName:   name,
-		ReleaseName: name,
+		ChartName:   chartName,
+		ReleaseName: releaseName,
```

Comment:
It seems like the added argument is still the same correct? Does this affect anything on the test end or just another way of adding the same parameter?

Reply:
this was a difference I found if the chart is installed manually vs. via automation. Automation seems to have chartName == releaseName. But manually deploying some charts like vsphere CPI / CSI sets the releaseName = chartName - `rancher-`.  which was causing problems with the upgrade. Since this involves a provisioning test, I wanted to update this so that the user could provide a cluster that was provisioned manually and/or via automation. This was the only way I could see to do that dynamically.